src/io/sampling.jl

```diff
@@ -65,24 +65,68 @@ end
 
 
 """
-    sample(dom::Domain, n::Int, sampler=SobolSample())::DataFrame
+    sample(dom::Domain, n::Int)::DataFrame
 
-Create samples using provided sampler, and rescale to distribution defined in the model spec.
+Create samples and rescale to distribution defined in the model spec.
 
 Notes:
 - assumes all parameters are independent.
 
 # Arguments
 - dom : Domain
 - n : Int
-- sampler : Sampling method
 """
-function sample(dom::Domain, n::Int, sampler=SobolSample(); supported_dists=Dict(
+function sample(dom::Domain, n::Int, sampler=SobolSample())::DataFrame
+    n > 0 ? n : throw(DomainError(n, "`n` must be > 0"))
+
+    spec = model_spec(dom)
+    df = sample(spec, n, sampler)
+
+    # Adjust samples for discrete values using flooring trick
+    # Ensure unguided scenarios do not have superfluous parameter values
+    return adjust_samples(dom, spec, df)
+end
+
+"""
+    sample(dom::Domain, n::Int, component::Type)::DataFrame
+
+Create samples and rescale to distribution defined in the model spec.
+
+Notes:
+- assumes all parameters are independent.
+
+# Arguments
+- dom : Domain
+- n : Int
+- component : Type, e.g. Criteria
+"""
+function sample(dom::Domain, n::Int, component::Type, sampler=SobolSample())::DataFrame
+    n > 0 ? n : throw(DomainError(n, "`n` must be > 0"))
+
+    spec = component_params(dom.model, component)
+    df = sample(spec, n, sampler)
+    process_inputs!(spec, df)
+
+    # Adjust samples for discrete values using flooring trick
+    # Ensure unguided scenarios do not have superfluous parameter values
+    return df
```

Comment:
No parameter adjustment or transform to specified distributions? Or is this handled elsewhere?

Reply:
It's done above this comment, so I'llmove the comment back above